pkg/k8s/apis/cilium.io/v2/cclrp_types.go

```diff
@@ -0,0 +1,52 @@
+//  Copyright 2020 Authors of Cilium
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package v2
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:singular="ciliumclusterwidelocalredirectpolicy",path="ciliumclusterwidelocalredirectpolicies",scope="Cluster",shortName={cclrp}
+
+// CiliumClusterwideLocalRedirectPolicy is a Kubernetes Custom Resource that is a
+// cluster-scoped version of CiliumLocalRedirectPolicy.
+type CiliumClusterwideLocalRedirectPolicy struct {
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	// +deepequal-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	CiliumLocalRedirectPolicy `json:",inline"`
```

Comment:
In CCNP, we embed a pointer of CNP. I'm not sure if we need to do that, or if that matters at all here. @aanm Was there a reason we embedded a pointer in CCNP?

Reply:
Should be a pointer. Having an embedded pointer allow us to use the structure as an interface.
```golang
package main

import (
	"fmt"
)

type Person struct {
	name string
}

func (p *Person) Name() string {
	return fmt.Sprintf("My name is %s", p.name)
}

type Student struct {
	Person
}

type StudentWithPointer struct {
	*Person
}

type NameTeller interface {
	Name() string
}

func main() {
	//s := Student{Person{name: "Abe"}}
	//nt := NameTeller(s)   -> Does not work
	//fmt.Println(nt.Name())

	swp := StudentWithPointer{&Person{name: "Abe Pointer"}}
	ntSwp := NameTeller(swp)
	fmt.Println(ntSwp.Name())
}
```

Reply:
@aanm Umm... I think the commented out code in your example doesn't work because you have defined `Name()` on a pointer type receiver, and passing the student object as a value. If we pass reference to a struct, it works - https://play.golang.org/p/yu3p97VInW0
Let me know if there is more to it. Thanks @christarazi for bringing this up. I also noticed that we were embedding pointer to CNP in CCNP, but not for CEP, etc.

Reply:
@aditighag yes that is correct but we usually have all receivers set with pointers.